src/core/qgsvectorlayerlabeling.cpp

```diff
@@ -88,3 +92,469 @@ QgsVectorLayerSimpleLabeling *QgsVectorLayerSimpleLabeling::create( const QDomEl
 
   return new QgsVectorLayerSimpleLabeling( QgsPalLayerSettings() );
 }
+
+QPointF quadOffsetToSldAnchor( QgsPalLayerSettings::QuadrantPosition quadrantPosition )
+{
+  double quadOffsetX = 0.5, quadOffsetY = 0.5;
+
+  // adjust quadrant offset of labels
+  switch ( quadrantPosition )
+  {
+    case QgsPalLayerSettings::QuadrantAboveLeft:
+      quadOffsetX = 1;
+      quadOffsetY = 0;
+      break;
+    case QgsPalLayerSettings::QuadrantAbove:
+      quadOffsetX = 0.5;
+      quadOffsetY = 0;
+      break;
+    case QgsPalLayerSettings::QuadrantAboveRight:
+      quadOffsetX = 0;
+      quadOffsetY = 0;
+      break;
+    case QgsPalLayerSettings::QuadrantLeft:
+      quadOffsetX = 1;
+      quadOffsetY = 0.5;
+      break;
+    case QgsPalLayerSettings::QuadrantRight:
+      quadOffsetX = 0;
+      quadOffsetY = 0.5;
+      break;
+    case QgsPalLayerSettings::QuadrantBelowLeft:
+      quadOffsetX = 1;
+      quadOffsetY = 1;
+      break;
+    case QgsPalLayerSettings::QuadrantBelow:
+      quadOffsetX = 0.5;
+      quadOffsetY = 1;
+      break;
+    case QgsPalLayerSettings::QuadrantBelowRight:
+      quadOffsetX = 0;
+      quadOffsetY = 1.0;
+      break;
+    case QgsPalLayerSettings::QuadrantOver:
+    default:
```

Comment:
Drop the default here - as the earlier discussion was about, it's better to just explicitly list every possible enum value and let the compiler throw a warning when a new value is added that isn't handled here.